src/main/java/com/hr/app/jpaexample/controller/DepartmentController.java

```diff
@@ -27,6 +24,9 @@ public List<Department> getAllDepartments() {
         return departmentService.getAllDepartments();
     }
 
+    @PostMapping(value = "/insert",consumes = "application/json",produces = "application/json")
+    public Department insertrow(@RequestBody Department dep){return departmentService.insertrow(dep);}
+
```

Comment:
Refer this API design document: https://blog.stoplight.io/crud-api-design

@PostMapping
public Department **createDepartment**(@RequestBody **DepartmentDto** departmentDto)

@PostMapping will have by default consumes and produce as application/json only

1) insertRow is generic name, you need to have specific naming conventions should be used whatever you are doing should be self explanatory by your method name example: **createDepartment**
2) Dempartment we already have entity name we should not use same entity class for request object, instead you should create new **DepartmentRequest** and use Mapper to copy value from DepartmentRequest object to Demaprtment entity object. and in service class method name should be **saveDepartment**  or **createDepartment** not the insert row